solr/core/src/java/org/apache/solr/handler/admin/ConfigSetsHandler.java

```diff
@@ -170,10 +170,14 @@ private void handleConfigUploadRequest(SolrQueryRequest req, SolrQueryResponse r
 
     boolean overwritesExisting = zkClient.exists(configPathInZk, true);
 
-    if (overwritesExisting && !req.getParams().getBool(ConfigSetParams.OVERWRITE, false)) {
-      throw new SolrException(ErrorCode.BAD_REQUEST,
-          "The configuration " + configSetName + " already exists in zookeeper");
-    }
+    // Get upload parameters
+    String singleFilePath = req.getParams().get(ConfigSetParams.FILE_PATH, "");
+    boolean allowOverwrite = req.getParams().getBool(ConfigSetParams.OVERWRITE, false);
+    // Cleanup is not allowed while using singleFilePath upload
+    boolean cleanup = singleFilePath.isEmpty() && req.getParams().getBool(ConfigSetParams.CLEANUP, false);
```

Comment:
should we error instead of silently ignoring the `cleanup` param? it defaults to `false`, so someone must have explicitly set it to `true`

Reply:
Added error handling for that and cases where a bad filePath is given.